fixedobject.go

```diff
@@ -0,0 +1,267 @@
+package schemer
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io"
+	"reflect"
+	"strconv"
+)
+
+type ObjectField struct {
+	Aliases []string
+	Schema  Schema
+}
+
+type FixedObjectSchema struct {
+	SchemaOptions
+	Fields []ObjectField
+}
+
+func (s *FixedObjectSchema) DefaultGOType() reflect.Type {
+	type tmpStruct struct {
+	}
+	var t tmpStruct
+	return reflect.TypeOf(t)
+}
+
+func (s *FixedObjectSchema) MarshalJSON() ([]byte, error) {
+	tmpMap := make(map[string]interface{}, 3)
+	tmpMap["type"] = "object"
+	tmpMap["nullable"] = strconv.FormatBool(s.SchemaOptions.Nullable)
+
+	var fieldMap []map[string]interface{}
+
+	for i := range s.Fields {
```

Comment:
A few things here:

- `fieldMap` should be an array of fields.  Each `ObjectField` could implement `MarshalJSON` to more elegantly solve this problem.
- ObjectField's `MarshalJSON` could work like this:
  - We ultimately return a JSON object containing `name` field, `type`, and other fields related to the schema
  - We could start by marshalling the `ObjectField.Schema`
  - As a bit of a hack, we can remove the trailing `}` from the returned `[]byte` and append the `name` field
  - The `name` field should be:
    - a string if only one alias is provided
    - an array if multiple aliases are provided